lib/services/component_manager.ts

```diff
@@ -183,7 +207,7 @@ export class SNComponentManager extends PureService {
     super.deinit();
     this.streamObservers.length = 0;
     this.contextStreamObservers.length = 0;
-    this.activeComponents.length = 0;
+    this.activeComponents.clear();
```

Comment:
`activeComponents ` should just be nulled out. We don't need a recycle method. The story is that `deinit` used to be a method for recycling a service, but then we moved towards complete nuking of a service. So you could do away with the clear function altogether and just set activeComponents to undefined.